### DISCUSSION TITLE
Maybe other Repl UI is a better option than Jupyer?

### DISCUSSION THREAD

**Discussion by baiguoname** [sigmaSd/IRust#112] [General]
Recently, I build a website, in which the user can input text, which will be evaluated by irust_repl, and than show the output. The process works fluently, just like python in Jupyter. 
So I thought, if it is difficult to embed irust in Jupyter, is it a better option to embed irust in its own Repl UI. For example, there is a Repl UI named [Pluto ](https://github.com/fonsp/Pluto.jl) in [Julia](https://julialang.org/), which is more powerful than Jupyter. Is it possible that irust make its own Pluto?

**Comment by sigmaSd** [comment]
That's cool but it seems like a decent amount of work
Actually integration to jupyter using the python bindings is not hard I'm just not personally that motivated to do it, and If we want to go the 100% rust way we can copy from excvr (but I don't think its worth it)
My own priorities in irust is rewriting it from the ground up to fix all  the hacks it picked with the time, but I haven't got to it yet

**Comment by actuday6418** [comment]
@sigmaSd when you say

> rewriting it from the ground up to fix all the hacks it picked with the time

Could you please expand a little more on how the rewrite is planned and what it is intended to fix? It would be really cool if we could just write a fully interpreted language that is a strict superset of Rust in syntax and semantics but does all the enforcement at runtime.

**Reply by sigmaSd** [reply]
> It would be really cool if we could just write a fully interpreted language that is a strict superset of Rust in syntax and semantics but does all the enforcement at runtime.

This is a rather big project, I don't think its belongs here, because rust is not a simple language to parse or evaluate, There is https://github.com/rust-lang/rustc_codegen_cranelift/blob/master/docs/usage.md#jit-mode btw It might fit in , in this idea somehow

**Reply by sigmaSd** [reply]
The rewrite is  mostly about terminal manipulation, currently its very hacky and have a lot of edge cases, because I wrote all of it by hand, using a higher level crate like reedline or readline would make things much easier https://github.com/sigmaSd/IRust/issues/104  

Also seeing stuff like claude code, gemini cli, I see that you can take terminal manipulation very far, I don't know if rust have something like https://www.npmjs.com/package/ink Though this maybe is too much, and one of the above crates is enough

The rest I guess is just codebase refactoring
